### PR TITLE
Add Experimental Relationship Counter API

### PR DESCRIPTION
**Pull request by josephschorr**
See #1860

### CODE REVIEW

File: internal/datastore/crdb/migrations/zz_migration.0006_add_relationship_counters_table.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	"github.com/jackc/pgx/v5"
+)
+
+const (
+	addRelationshipCountersTableQuery = `
+		CREATE TABLE relationship_counter (
+			name TEXT PRIMARY KEY,
+			serialized_filter BYTEA NOT NULL,
+			current_count BIGINT NOT NULL DEFAULT 0,
+			updated_at_timestamp DECIMAL DEFAULT NULL
```
[the remark this answers is not in the document]
Comment (josephschorr) [reply internal/datastore/crdb/migrations/zz_migration.0006_add_relationship_counters_table.go:15]: Its a Hybrid Logical Clock, so we need the DECIMAL to store it

File: internal/datastore/crdb/migrations/zz_migration.0006_add_relationship_counters_table.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+		);
+	`
+)
+
+func init() {
+	err := CRDBMigrations.Register("add-relationship-counters-table", "remove-stats-table", addRelationshipCountersTable, noAtomicMigration)
```
[the remark this answers is not in the document]
Comment (josephschorr) [reply internal/datastore/crdb/migrations/zz_migration.0006_add_relationship_counters_table.go:21]: I think so? This has been present for quite a while

File: internal/services/v1/experimental.go [hunk 3]
```diff
@@ -634,6 +637,72 @@ func (es *experimentalServer) ExperimentalDependentRelations(ctx context.Context
...
+	}
+
+	ds := datastoremw.MustFromContext(ctx)
+	headRev, err := ds.HeadRevision(ctx)
+	if err != nil {
+		return nil, shared.RewriteErrorWithoutConfig(ctx, err)
+	}
+
+	snapshotReader := ds.SnapshotReader(headRev)
+	count, err := snapshotReader.CountRelationships(ctx, req.Name)
```
[the remark this answers is not in the document]
Comment (josephschorr) [reply internal/services/v1/experimental.go:691]: Not currently; its designed so we can layer the counting service on top as the followup

**Comment by josephschorr** [comment]
Updated

File: internal/datastore/crdb/migrations/zz_migration.0006_add_relationship_counters_table.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (ecordell) [reply internal/datastore/crdb/migrations/zz_migration.0006_add_relationship_counters_table.go:21]: yep it's already done

**Review by ecordell — APPROVED** [review]

**Pull request merged**